scipy/optimize/_root.py

```diff
@@ -637,3 +639,86 @@ def _root_krylov_doc():
             See `scipy.sparse.linalg.lgmres` for details.
     """
     pass
+
+
+def root_scalar(f, a, b, args=(), method='brentq', xtol=_xtol, rtol=_rtol,
+                maxiter=_iter, full_output=False, disp=True):
+    """
+    Find a root of a function in a bracketing interval.
+
+    Parameters
+    ----------
+    f : function
+        Python function returning a number. Must be continuous and `f(a)` and
+        `f(b)` must have opposite signs.
+    a : number
+        One end of the bracketing interval `[a,b]`.
+    b : number
+        The other end of the bracketing interval `[a,b]`.
+    args : tuple, optional
+        Extra arguments for the function `f`. In the code `f` is called by
+        ``apply(f, (x)+args)``.
+    method: str, optional
+        Type of solver. Should be one of
+
+            - 'brentq'               (see :func:`brentq`)
+            - 'brenth'               (see :func:`brenth`)
+            - 'ridder'               (see :func:`ridder`)
+            - 'bisect'               (see :func:`bisect`)
+
+    xtol : number, optional
+        The routine converges when a root is known to lie within `xtol` of the
+        value return. Should be >= 0.  The routine modifies this to take into
+        account the relative precision of doubles.
+    rtol : number, optional
+        The routine converges when a root is known to lie within `rtol` times
+        the value returned of the value returned. Should be >= 0. Defaults to
+        ``np.finfo(float).eps * 2``.
+    maxiter : number, optional
+        If convergence is not achieved in `maxiter` iterations, an error is
+        raised.  Must be >= 0.
+    full_output : bool, optional
+        If `full_output` is False, the root is returned.  If `full_output` is
+        True, the return value is ``(x, r)``, where ``x`` is the root,
+        and ``r`` is a ``RootResults`` object.
+    disp : bool, optional
+        If ``True``, raise RuntimeError if the algorithm didn't converge.
+
+    Returns
+    -------
+    x : float
+        Zero of `f` between `a` and `b`.
+    r : RootResults (present if ``full_output = True``)
+        Object containing information about the convergence.  In particular,
+        ``r.converged`` is ``True`` if the routine converged.
+
```

Comment:
Maybe we should return an instance of `OptimizeResult` instead, to be consistent with `root` and `minimize`.